lib/widgets/message_list.dart

```diff
@@ -354,62 +376,65 @@ class MessageItem extends StatelessWidget {
-      restBorderColor = _kDmRecipientHeaderColor;
-    } else {
-      throw Exception("impossible message type: ${message.runtimeType}");
-    }
-
-    // This 3px border seems to accurately reproduce something much more
-    // complicated on web, involving CSS box-shadow; see comment below.
-    final recipientBorder = BorderSide(color: highlightBorderColor, width: 3);
-    final restBorder = BorderSide(color: restBorderColor, width: 1);
-    var borderDecoration = ShapeDecoration(
-      // Web actually uses, for stream messages, a slightly lighter border at
-      // right than at bottom and in the recipient header: black 10% alpha,
-      // vs. 88% lightness.  Assume that's an accident.
-      shape: Border(
-        left: recipientBorder,
-        right: restBorder,
-        bottom: item.isLastInBlock ? restBorder : BorderSide.none,
-      ));
-
     return StickyHeaderItem(
       allowOverflow: !item.isLastInBlock,
       header: RecipientHeader(message: message),
-      child: Column(children: [
-        DecoratedBox(
-          decoration: borderDecoration,
-          child: MessageWithPossibleSender(item: item)),
-        if (trailing != null && item.isLastInBlock) trailing!,
-      ]));
-
-    // Web handles the left-side recipient marker in a funky way:
-    //   box-shadow: inset 3px 0px 0px -1px #c2726a, -1px 0px 0px 0px #c2726a;
-    // (where the color is the stream color.)  That is, it's a pair of
-    // box shadows.  One of them is inset.
-    //
-    // At attempt at a literal translation might look like this:
-    //
-    // DecoratedBox(
-    //   decoration: ShapeDecoration(shadows: [
-    //     BoxShadow(offset: Offset(3, 0), spreadRadius: -1, color: highlightBorderColor),
-    //     BoxShadow(offset: Offset(-1, 0), color: highlightBorderColor),
-    //   ], shape: Border.fromBorderSide(BorderSide.none)),
-    //   child: MessageWithSender(message: message)),
-    //
-    // But CSS `box-shadow` seems to not apply under the item itself, while
-    // Flutter's BoxShadow does.
+      child: _UnreadMarker(
+        isRead: message.flags.contains(MessageFlag.read),
+        child: Column(children: [
+          MessageWithPossibleSender(item: item),
+          if (trailing != null && item.isLastInBlock) trailing!,
+        ])));
+  }
+}
+
+/// Widget responsible for showing the read status of a message.
+// Mobile currently uses a single inset box-shadow:
+//   box-shadow: inset 4px 0 hsl(222, 99%, 69%);
+// The animation in mobile has a 1s delay, and a 0.3s
+// transition using the ease-out curve.
+// See zulip-mobile:src/webview/static/base.css .
+//
+// Modern web uses a block element will a linear-gradient
+// background fill:
+//   linear-gradient(90deg, hsl(217deg 64% 59%) 30%, hsl(217deg 64% 59% / 0%))
+// Transition is 0.3s for appearing and 2s for disappearing,
+// both with an ease-out curve.
+// See zulip:web/styles/message_row.css .
+//
+// Current design work is ongoing and values used here come
```

Comment:
A remark like "current design work is ongoing" is something that typically won't still be true when we merge a change. So that means it generally doesn't belong in the code of the PR, which is a proposal for code to merge. Instead it's a good remark to make in the PR thread, e.g. in the form of an inline comment (a GitHub "review" comment).